types/regenerator-runtime/index.d.ts

```diff
@@ -0,0 +1,236 @@
+
+    /**
+     * @param record The completion record.
+     * @param afterLoc The location to resume the generator at, only used by normal completions.
+     */
+    complete(record: Readonly<CompletionRecord>, afterLoc?: ContextLocation): unknown;
+
+    /**
+     * Used to signify the end of a finally block.
+     *
+     * @param finallyLoc The label of the beginning of the finally block.
+     */
+    finish(finallyLoc: number): unknown;
+
+    /**
+     * Used to obtain the exception that was thrown in the associated try block.
+     *
+     * @param tryLoc The label of the beginning of the try block.
+     */
+    catch(tryLoc: number): unknown;
+
+    /**
+     * @param iterable The iterable to delegate to.
+     * @param resultName The name of the property to assign to on this context.
+     * @param nextLoc The label of the location where to resume iteration.
+     */
+    delegateYield(
+        iterable: { [Symbol.iterator](): Iterator<TYield, unknown, unknown> },
+        resultName: string,
+        nextLoc: ContextLocation,
+    ): unknown;
+
+    [
+        /**
+         * Expects properties to match `/^t[+-]?\d*(?:(?<=\d)\.\d*|\.\d+)?(?:e[+-]?\d+)?$/`.
+         */
+        temp: string
+    ]: any;
+}
+
+export function wrap<T = undefined, TYield = unknown, TReturn = unknown, TNext = unknown>(
+    innerFn: InnerFunction<T, TYield, TReturn, TNext>,
+    // tslint:disable-next-line: ban-types
+    outerFn?: Function | null,
+    self?: T,
+    tryLocsList?: TryLocationsList,
+): Generator<TYield, TReturn, TNext>;
+
+export interface ResolvablePromiseConstructorLike extends PromiseConstructorLike {
+    resolve<T = void>(value?: T): PromiseLike<T>;
+}
+
+export class AsyncIterator<TYield = unknown, TReturn = unknown, TNext = unknown>
+    implements AsyncGenerator<TYield, TReturn, TNext> {
+    constructor(
+        generator: Generator<
+            TYield | PromiseLike<TYield> | awrap<unknown>,
+            TReturn | PromiseLike<TReturn> | awrap<unknown>
+        >,
+        PromiseImpl: ResolvablePromiseConstructorLike,
+    );
+
+    next(value: TNext): Promise<IteratorResult<TYield, TReturn>>;
+    return(value: TReturn | PromiseLike<TReturn>): Promise<IteratorResult<TYield, TReturn>>;
+    throw(e: any): Promise<IteratorResult<TYield, TReturn>>;
+
+    [Symbol.asyncIterator](): this;
+}
+
+export function async<T = undefined, TYield = unknown, TReturn = unknown>(
+    innerFn: InnerFunction<T, TYield, TReturn>,
+    outerFn: GeneratorFunction,
+    self?: T,
+    tryLocsList?: TryLocationsList,
+    PromiseImpl?: ResolvablePromiseConstructorLike,
+): AsyncIterator<
+    TYield extends PromiseLike<infer Await> ? Await : Exclude<TYield, awrap<unknown>>,
+    TReturn extends PromiseLike<infer Await> ? Await : Exclude<TReturn, awrap<unknown>>
+>;
+export function async<T = undefined, TReturn = unknown>(
+    innerFn: InnerFunction<T, unknown, TReturn>,
+    // tslint:disable-next-line: ban-types
+    outerFn?: Function | null,
```

Comment:
Do not disable this error:: Fix this instead by changing to `() => void`

Reply:
That signature is incorrect, as it won’t accept class constructors and anything that takes at least one required parameter.

The `Function` type most closely describes what the `wrap` and `async` functions need: <https://github.com/facebook/regenerator/blob/0566ce53c67b5fe1de13b47ed4db00570de7d9f6/packages/regenerator-runtime/runtime.js#L39>.